### DISCUSSION TITLE
Clair V4 (4.6.1) failed to connect to the PostgreSQL database

### DISCUSSION THREAD

**Discussion by Oliviertjeh** [quay/clair#1786] [Design]
My name is Olivier, and I am a student working on a project with Clair V4. Unfortunately, I have problems connecting Clair V4 (V4.6.1) to the PostgreSQL database. And I hope someone can help me explain what I am doing wrong.

**The error I am getting:**
{"level":"error","component":"main","error":"service initialization failed: failed to initialize indexer: failed to create ConnPool: failed to connect to `host=postgres user=postgres database=clair`: hostname resolving error (lookup postgres on 192.168.190.2:53: no such host)","time":"2023-06-16T11:52:34Z","message":"fatal error"}

**The environment I am working on:**
-Locally on VMware with an internet connection
-VMware machine
-Ubuntu 22.0.4
-Using Podman

**The following steps have I done:**
1.	Postgres install
`podman pull postgres:11.5`

```
podman run -d \
  --name clairv4-postgres \
  -p 5432:5432 \
  -e POSTGRES_USER=postgres \
  -e POSTGRES_DB=clair \
  -e POSTGRES_PASSWORD=postgres \
  -e PGDATA=/etc/postgres/data \
  -v /home/olivier/poc/postgres:/etc/postgres \
  postgres:11.5
```

**2.	Clair V4 install**
`podman pull quay.io/projectquay/clair:4.6.1`

```
podman run -d \
  --name clair \
  -p 8080:8080 \
  -v /home/olivier/poc/clair/config.yaml:/config/config.yaml \
  -v /home/olivier/poc/clair/db:/home/olivier/poc/clair/db \
  quay.io/projectquay/clair:4.6.1
```

**The config file I am using (config.yaml):**
```
introspection_addr: 0.0.0.0:8089
http_listen_addr: 0.0.0.0:8080
log_level: debug
indexer:
  connstring: 'postgresql://postgres:postgres@postgres/clair'
  scanlock_retry: 10
  layer_scan_concurrency: 5
  migrations: true
  airgap: false
matcher:
  indexer_addr: "localhost:8080"
  connstring: 'postgresql://postgres:postgres@postgres/clair'
  max_conn_pool: 100
  run: ""
  migrations: true
updater_sets:
    - "alpine"
    - "aws"
    - "debian"
    - "oracle"
    - "photon"
    - "pyupio"
    - "rhel"
    - "suse"
    - "ubuntu"
# tracing and metrics
trace:
  name: "jaeger"
  probability: 1
  jaeger:
    agent_endpoint: "localhost:6831"
    service_name: "clair"

metrics:
  name: "prometheus"
```

**Comment by hdonnay** [comment]
IIRC the container name is the DNS entry, so `clairv4-postgres` doesn't match up with `host=postgres`.
